complex/cargo/list/src/items.rs

```diff
@@ -38,11 +51,26 @@ impl Drop for Item {
     }
 }
 
+impl Item {
+    pub fn from_row(row: &Vec<TypedValue>) -> Option<Item> {
+        let item = Item{
+            id: row[0].clone().to_inner(),
+            uuid: row[1].clone().to_inner(),
+            name: row[2].clone().to_inner(),
+            due_date: row[3].clone().to_inner(),
+            completion_date: row[4].clone().to_inner(),
+            labels: vec![]
+        };
+        Some(item)
+    }
+}
+
 #[no_mangle]
 pub extern "C" fn item_new() -> *mut Item {
     let item = Item{
```

Comment:
I encourage you to have `Item` implement `Default`, and to use that instead of manually constructing one. But `Uuid::nil()` is perhaps not the right choice here. What's `item_new` for?

Reply:
It's a little bit historic. Originally the iOS app created a new blank item, filled in the details and saved the item by passing it back over the boundary. I know Grisha is changing this to take the properties as args for item_new and constructing the item from those values. I was trying to move this over to Mentat without fucking with the API too much to not screw over everyone else.